### DISCUSSION TITLE
Certbot & Owncast

### DISCUSSION THREAD

**Discussion by thraeryn** [owncast/owncast#571] [Q&A] [answered]
There's no tutorial for installing certbot keys on Owncast. Should I have certbot generate standalone keys? How can I install them once they're generated?

**Comment by gabek** [comment]
Most people have their own SSL solutions, and we're not forcing a specific implementations.  There's a lot of use of Caddy, nginx, traefik, etc as a reverse proxy with Owncast and that's the route I think is best suited.

**Comment by thraeryn** [comment]
Rough. I'm an out-of-the-box user who's perfectly good at reading a certbot
tutorial and copying a console line for an install script, editing a config
file,

but I don't really know about reverse proxies, and I'm not certain I'm
ready to install a web server in addition to this almost exclusively so I
can install a cert.

I'll live without until I take an afternoon to learn something new and bash
my head against implementation for a few hours, it seems.

**Comment by jeyemwey** [comment, marked as answer]
The overhead of having an https server right inside Owncast is not really worth the effort right now, especially since most people use Owncast next to other web software on one host. I suggest you to look into [Caddy](https://caddyserver.com). My server configuration for caddy is this:

```
stream.iamjannik.me {
  proxy / localhost:8080 {
    transparent
    websocket
  }
}
```

Caddy will automatically create a Let’s Encrypt certificate for this exact domain and keep it renewed just in time.

**Reply by geekgonecrazy** [reply]
yeah if you are looking for quickest out of the box i'd go for this.  Alternatively can even download caddy and start up with single command and have going.

```
./caddy reverse-proxy --from live.yourdomain.com --to 127.0.0.1:8080
```

Just have dns already pointing at your machine and thats it.  

Soon we should have documentation for various options.  Its being worked on

**Reply by jeyemwey** [reply]
Update: The Caddyfile above only seems to work with Caddy 1. For caddy 2, check out this Caddyfile:

```
stream.example.com {
    encode gzip
    reverse_proxy 127.0.0.1:8080
}
```

**Reply by dwopetweens** [reply]
can you help me fix errors on my caddy server ?

**Comment by thraeryn** [comment]
Thanks. That's really useful info. I appreciate your taking the time to
give me more assistance, and I look forward to the upcoming docs.

**Comment by RobertoJBeltran** [comment]
I definitely agree that it makes sense to just use the tools already out there, and that I should probably already know some of these things. Having some of the real basic stuff even mentioned in the docs would have saved me an hour or two though. Here's what I've gathered in case it helps anyone passing through until the new docs are up

https://gitlab.com/libremiami/libremiami/-/issues/42#note_503367630
